PluginsAndFeatures/azure-toolkit-for-rider/azure-intellij-plugin-bicep/src/main/kotlin/com/microsoft/azure/toolkit/intellij/bicep/lsp/LsInfrastructure.kt

```diff
@@ -0,0 +1,67 @@
+package com.microsoft.azure.toolkit.intellij.bicep.lsp
+
+import com.microsoft.azure.toolkit.intellij.bicep.BicepBundle
+import com.intellij.execution.configurations.GeneralCommandLine
+import com.intellij.execution.util.ExecUtil
+import com.intellij.openapi.diagnostic.thisLogger
+import com.intellij.openapi.util.NlsSafe
+import com.intellij.openapi.util.SystemInfo
+import com.intellij.openapi.util.io.toNioPathOrNull
+import com.intellij.util.PathUtil
+import com.intellij.util.concurrency.annotations.RequiresBackgroundThread
+import com.intellij.util.concurrency.annotations.RequiresReadLockAbsence
+import org.jetbrains.annotations.Nls
+import java.nio.file.Path
+import kotlin.io.path.absolutePathString
+import kotlin.io.path.exists
+
+internal sealed class LsInfrastructure(protected val urlTemplate: String) {
+  data object BicepLS : LsInfrastructure("https://github.com/Azure/bicep/releases/download/%s/bicep-langserver.zip") {
+    override val formattedUrl: String
+      get() {
+        val supportedVersion = "v0.31.92"
```

Comment:
It's a pretty outdated version. Maybe we could use the latest one? Anyway, we won't have enough resources to test every version